### DISCUSSION TITLE
Radiomaster TX12 - 64KB vs 128KB difference?

### DISCUSSION THREAD

**Discussion by Miwer** [pascallanger/DIY-Multiprotocol-TX-Module#692] [Q&A] [answered]
Hi,

I am one of those unlucky to have a TX12 where the internal module is equipped with a C8 (64K) chip instead of a CB (128K) chip.

Is there any difference in the enabled protocols/features between the two versions, released for this module?
I'm trying to find out if it's worth the time of swapping out the C8 chip with a blank CB chip + programming to enable more features.

I already have a spare CB chip, and by looking at the [flash-multi](https://github.com/benlye/flash-multi/blob/master/linux/flash-multi) source, I managed to program the [StmMulti4in1_StickyDfu.bin](https://github.com/benlye/flash-multi/blob/master/linux/bootloader/StmMulti4in1_StickyDfu.bin) to 0x08000000 and [mm-stm-cc2500-aetr-v1.3.3.7.bin](https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/releases/download/v1.3.3.7/mm-stm-cc2500-aetr-v1.3.3.7.bin) to 0x08002000 using ST-Link programmer.

But now I wonder if it's even worth the hassle of desoldering the C8 chip and solder in the CB chip, if the C8 chip have plenty space available for the features of only CC2500.

Should I just stick with the C8 chip ?

[edit: corrected chip addresses]

**Comment by pascallanger** [comment, marked as answer]
Most if not all of the TX 12 are shipped with a C8. There is no need to do anything, just load the 64KB firmware and you are good to go.

**Reply by Miwer** [reply]
@pascallanger thank you for replying.

So just to confirm, there is no features/protocols disabled in the 64KB version due to space constraints, correct?

**Reply by pascallanger** [reply]
Nothing is disabled, for now at least.

**Comment by djscratch98** [comment]
I have the C8 chip and have a heck of a time fixing a bad flash. I am no programmer but tried to compile using the Arduino IDE ...LOL NO luck...the lingo got me confused. I was able to do something because it said to // Comment out some of the protocols you don't need...well I must have took some out but now it shows up some of the protocols and then some are just numbers??? It's screwed. I think I need the bootloader compiled with a 64kb version programmed, I also tried to use the Flash multi tool to try a flash and it was worse than the compiler. I am confused and hope someone on the RCGroups forum can program me what I need and send it in a bin file. I have tried old, new 64kb, 128kb versions and they all are not working using the flash multi tool. My questions are on the RCGroups link here.  https://www.rcgroups.com/forums/showthread.php?4230837-Radiomaster-Tx12-Multi-Module-FAIL%21%21%21-Need-re-flashed-bootloader

**Reply by djscratch98** [reply]
Nevermind I fixed the issue!
